doc/source/tutorials/transpiler.rst

```diff
@@ -0,0 +1,152 @@
+In this tutorial we will demonstrate how the user can modify this process for custom applications.
+
+The ``transpiler`` and ``compiler`` objects used when executing a circuit are attributes of :class:`qibolab.backends.QibolabBackend`.
+Creating an instance of the backend provides access to these objects:
+
+.. code-block:: python
+
+    from qibolab.backends import QibolabBackend
+
+    backend = QibolabBackend(platform="single_qubit")
+    print(backend.transpiler)
+    print(backend.compiler)
+
+The transpiler is responsible for transforming the circuit to respect the chip connectivity and native gates,
+while the compiler transforms the circuit to the equivalent pulse sequence.
+The user can modify these attributes before executing a circuit.
+For example:
+
+.. code-block:: python
+
+    from qibo import gates
+    from qibo.models import Circuit
+    from qibolab.backends import QibolabBackend
+
+    # define circuit
+    circuit = Circuit(1)
+    circuit.add(gates.U3(0, 0.1, 0.2, 0.3))
+    circuit.add(gates.M(0))
+
+    backend = QibolabBackend(platform="single_qubit")
+    # disable the transpiler
+    backend.transpiler = None
+
+    # execute circuit
+    result = backend.execute_circuit(circuit, nshots=1000)
+
+completely disables the transpilation steps. In this case the circuit will be sent directly to the compiler, to be compiled to a pulse sequence.
+Instead of completely disabling, custom transpilation steps can be given:
+
+.. code-block:: python
+
+    from qibolab.backends import QibolabBackend
+    from qibolab.native import NativeType
+    from qibolab.transpilers.gate_decompositions import NativeGates
+
+    backend = QibolabBackend(platform="single_qubit")
+    backend.transpiler = NativeGates(two_qubit_natives=NativeType.CZ)
+
+
+Now circuits will only be transpiled to native gates, without any connectivity matching steps.
+The :class:`qibolab.transpilers.gate_decompositions.NativeGates` transpiler used in this example assumes Z, RZ, GPI2 or U3 as the single-qubit native gate, and supports CZ and iSWAP as two-qubit natives.
+In this case we restricted the two-qubit gate set to CZ only.
+
+Multiple transpilation steps can be implemented using the :class:`qibolab.transpilers.pipeline.Pipeline`:
+
+.. code-block:: python
+
+    from qibolab.native import NativeType
+    from qibolab.transpilers.pipeline import Pipeline
+    from qibolab.transpilers.star_connectivity import StarConnectivity
+    from qibolab.transpilers.gate_decompositions import NativeGates
+
+    backend = QibolabBackend(platform="single_qubit")
+    backend.transpiler = Pipeline(
+        [
+            StarConnectivity(middle_qubit=2),
+            NativeGates(two_qubit_natives=NativeType.CZ),
+        ]
+    )
+
+In this case circuits will first be transpiled to respect the 5-qubit star connectivity, with qubit 2 as the middle qubit. This will potentially add some SWAP gates. Then all gates will be converted to native.
```

Comment:
The example idea looks good, however you are using a `single_qubit` platform which might sound strange when compiling for 5 qubits.

Reply:
Good point. I changed the name to `my_platform` so that it does not specify a specific qubit configuration and is also TII agnostic.